src/parser.cc

```diff
@@ -1358,12 +1358,18 @@ void addNestedModel(ElementPtr _sdf, ElementPtr _includeSDF, Errors &_errors)
   for (std::map<std::string, std::string>::iterator iter = replace.begin();
        iter != replace.end(); ++iter)
   {
-    replace_all(str, std::string("\"") + iter->first + "\"",
-                std::string("\"") + iter->second + "\"");
-    replace_all(str, std::string("'") + iter->first + "'",
-                std::string("'") + iter->second + "'");
-    replace_all(str, std::string(">") + iter->first + "<",
-                std::string(">") + iter->second + "<");
+    std::string oldName(iter->first);
+    std::string nameWithNestedPrefix(iter->second);
+    replace_all(str, std::string("\"") + oldName + "\"",
+                     std::string("\"") + nameWithNestedPrefix + "\"");
+    replace_all(str, std::string("'") + oldName + "'",
+                     std::string("'") + nameWithNestedPrefix + "'");
+    replace_all(str, std::string(">") + oldName + "<",
+                     std::string(">") + nameWithNestedPrefix + "<");
+    // Deal with nested model inside other nested model and already with
+    // ::namespace:: entries in the name
+    replace_all(str, std::string(">") + oldName + "::",
```

Comment:
In https://github.com/ignitionrobotics/sdformat/pull/597, I see tests related to this in `test/integration/nested_model.cc`. Should we not merge those forward as well?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/738/commits/38be09c68fd15e308710717e02f128cca9a1d4da